test/utility/iterator.cpp

```diff
@@ -157,6 +158,19 @@ CONCEPT_ASSERT(IndirectlyMovable<int const *, int *>());
 CONCEPT_ASSERT(!IndirectlySwappable<int const *, int const *>());
 CONCEPT_ASSERT(!IndirectlyMovable<int const *, int const *>());
 
+namespace boost
+{
+    struct my_dummy{}; // just to have a type from boost namespace
+}
+
+// Regression test for https://github.com/ericniebler/range-v3/issues/845
+void test_845()
+{
+    std::list<boost::my_dummy> v = { boost::my_dummy{} };
+    auto itr = v.begin();
+    ranges::advance(itr, 1); // Should not create ambiguity
```

Comment:
This would be a much more useful test if we use the namespace `notboost`, and put a conflicting definition of `advance` in it.